src/traces/scatter/stack_defaults.js

```diff
@@ -0,0 +1,101 @@
+                // but for the editor and validate we want things self-consistent
+                // The full set of traces is used only to fix `fill` default if
+                // we find `orientation: 'h'` beyond the first trace
+                traces: [traceOut]
+            };
+            firstTrace = true;
+        }
+        // TODO: how is this going to work with groupby transforms?
+        // in principle it should be OK I guess, as long as explicit group styles
+        // don't override explicit base-trace styles?
+
+        var dflts = {
+            orientation: (traceOut.x && !traceOut.y) ? 'h' : 'v'
+        };
+
+        for(var i = 0; i < perStackAttrs.length; i++) {
+            var attr = perStackAttrs[i];
+            var attrFound = attr + 'Found';
+            if(!groupOpts[attrFound]) {
+                var traceHasAttr = traceIn[attr] !== undefined;
+                var isOrientation = attr === 'orientation';
+                if(traceHasAttr || firstTrace) {
+                    groupOpts[attr] = coerce(attr, dflts[attr]);
+
+                    if(isOrientation) {
+                        groupOpts.fillDflt = groupOpts[attr] === 'h' ?
+                            'tonextx' : 'tonexty';
+                    }
+
+                    if(traceHasAttr) {
+                        // TODO: this will show a value here even if it's invalid
+                        // in which case it will revert to default. Is that what we
+                        // want, or should we only take the first *valid* value?
+                        groupOpts[attrFound] = true;
+
+                        // TODO: should we copy the new value to the first trace instead?
+                        // if we did that, the editor could show controls in the
+                        // first trace always; otherwise it would only show controls
+                        // in a later trace, if that trace happens to already have a value...
+                        // I think it's probably better to ignore the editor for this
+                        // purpose though, as it's probably going to have to use some
+                        // special logic anyway, it will likely want to just
+                        // take the value out of layout._scatterStackOpts, set the
+                        // new value into the first trace, and clear all later traces.
+                        if(!firstTrace) {
+                            delete groupOpts.traces[0][attr];
+
+                            // orientation can affect default fill of previous traces
+                            if(isOrientation) {
+                                for(var j = 0; j < groupOpts.traces.length - 1; j++) {
+                                    var trace2 = groupOpts.traces[j];
+                                    if(trace2._input.fill !== trace2.fill) {
+                                        trace2.fill = groupOpts.fillDflt;
```

Comment:
I forgot if we discussed this before, but is there a use case for having a stacked area trace with a `fill` setting other than `tonexty` (`tonextx` for horizontal stacks)?

If not, we could skip `coerce('fill')` for stacked area traces and infer the `fill` value later on in scatter `plot.js` and `line_points.js` and :hocho: this hacky block.

Reply:
There's certainly a use case for no fill. Other than that no I don't see a use case, but I'm not sure that helps us remove this block if we still want to support `fill: 'none'`.

Reply:
> There's certainly a use case for no fill. 

Absolutely. Good point!

--------

Referecing

> mode is otherwise all about how to draw the series, not where to draw it... and the one bit of how that stacking impacts (fill) isn't even part of mode.

from https://github.com/plotly/plotly.js/pull/2960#issuecomment-418584839 (your reply to `mode: 'stack'`)

That _how_ vs _where_ argument makes sense to me :ok_hand: 

But it does suggest that the `fill` attribute isn't optimal. Here's a proposal for v2... maybe v3 :smirk: 

- :hocho: `fill: 'none'`
- add `'fill'` flag in scatter `mode`
- add that `'fill'` flag the `mode` default whenever `stackgroup` is set
- coerce scatter `fill` (that maybe we should call `fills` similar to the `markers` and `lines` attribute containers) only if `'fill'` is part of `mode`
- make `tonexty` the only valid `fills` value when `stackgroup` is set (`tonextx` for horizontal stacks). `Lib.log` something if some other value is set.

I think implementing this proposal would make this hacky logic obsolete.

Reply:
oh and by the way, I'm now convinced that adding a `'stack'` flag in mode isn't worth it :ok_hand: